solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -476,25 +601,72 @@ pub struct ChainWithVerifier<'info> {
 }
 
 #[derive(Accounts)]
-#[instruction(port_id: ibc::PortId, channel_id_on_b: ibc::ChannelId, hashed_base_denom: CryptoHash)]
-pub struct InitMint<'info> {
+pub struct SetupFeeCollector<'info> {
+    #[account(mut)]
+    fee_collector: Signer<'info>,
+
+    /// The account holding private IBC storage.
+    #[account(mut, seeds = [SOLANA_IBC_STORAGE_SEED],
+    bump)]
+    storage: Account<'info, storage::PrivateStorage>,
+}
+
+#[derive(Accounts)]
+pub struct CollectFees<'info> {
     #[account(mut)]
+    fee_collector: Signer<'info>,
+
+    /// The account holding private IBC storage.
+    #[account(mut, seeds = [SOLANA_IBC_STORAGE_SEED],
+    bump, has_one = fee_collector)]
```

Comment:
Uh? What’s the purpose of has_one here?

Reply:
only fee collector can call this method.